File: webserver/src/main/java/com/llh/webserver/service/SysUserService.java
package com.llh.webserver.service;

import com.llh.webserver.common.constant.AccountStatus;
import com.llh.webserver.model.SysUser;
import org.springframework.security.core.userdetails.UserDetailsService;

/**
 * SysUserService
 * <p>
 * CreatedAt: 2020-04-18 16:58
 *
 * @author llh
 */
public interface SysUserService extends BasicService<SysUser>, UserDetailsService {
    /**
     * 帐户锁定状态
     */
    int LOCKED_ACCOUNT = AccountStatus.LOCKED.ordinal();
    /**
     * 帐户移除状态
     */
    int REMOVED_ACCOUNT = AccountStatus.REMOVED.ordinal();
    /**
     * 帐户未激活状态
     */
    int INACTIVE_ACCOUNT = AccountStatus.INACTIVE.ordinal();

    /**
     * 帐户激活状态
     */
    int ACTIVE_ACCOUNT = AccountStatus.ACTIVE.ordinal();
}

File: server-kt[discard]/src/main/kotlin/com/llh/server/service/SysUserService.kt
package com.llh.server.service

import com.llh.server.common.constant.AccountStatus
import com.llh.server.model.SysUser
import com.llh.server.pojo.PageDTO
import com.llh.server.pojo.SimplePageQueryVO
import org.springframework.security.core.userdetails.UserDetailsService


/**
 * SysUserService
 *
 * CreatedAt: 2020-06-01 23:27
 *
 * @author llh
 */
interface SysUserService : BasicService<SysUser>, UserDetailsService {
    /**
     * 根据用户名[username]查找数据
     */
    fun findTopByUsername(username: String): SysUser?

    fun page(queryVO: SimplePageQueryVO<SysUser>): PageDTO<SysUser>

    val inactive: Int
        get() = AccountStatus.INACTIVE.ordinal
    val activation: Int
        get() = AccountStatus.ACTIVATION.ordinal
    val locked: Int
        get() = AccountStatus.LOCKED.ordinal
    val removed: Int
        get() = AccountStatus.REMOVED.ordinal
}
